src/idlcxx/src/streamers.c

```diff
@@ -152,48 +152,93 @@ process_case(
 {
   struct streams *streams = user_data;
   const idl_case_t* _case = (const idl_case_t*)node;
-  char *type = NULL;
+  char *type, *value = "";
   const char *name;
+  bool simple, single;
 
+  const char *readfmt;
   static const char writefmt[] =
-    "  write(instance.%s());\n"
-    "  break;\n";
-  static const char readfmt[] =
-    "{\n"
-    "  %1$s obj;\n"
-    "  read(str, obj);\n"
-    "  %2$s(obj, _disc_temp);\n"
-    "}\n"
-    "  break;\n";
+    "      write(str, instance.%s());\n"
+    "      break;\n";
   static const char movefmt[] =
-    "  move(str, instance.%s());\n"
-    "  break;\n";
+    "      move(str, instance.%s());\n"
+    "      break;\n";
   static const char maxfmt[] =
-    "{\n"
-    "  size_t pos = str.position();\n"
-    "  size_t alignment = str.alignment();\n"
-    "  max(str, instance.%s());\n"
-    "  union_max = std::max(str.position(),union_max);\n"
-    "  str.position(pos);\n"
-    "  str.alignment(alignment);\n"
-    "}\n";
+    "  {\n"
+    "    size_t pos = str.position();\n"
+    "    size_t alignment = str.alignment();\n"
+    "    max(str, instance.%s());\n"
+    "    if (union_max > str.position()) {\n"
```

Comment:
Should be `union_max > str.position()` since we want to select the largest, not the smallest.